### PR TITLE
Clarified documentation of mbedtls_ssl_setup

### PR DESCRIPTION
**Pull request by gilles-peskine-arm**
Note that the configuration structure must remain accessible. The
previous wording could have been taken as implying that it's ok to
change the structure but changes wouldn't be taken into account.

Also note that calling this function twice is not supported (it would
at least be a memory leak).

Internal ref: IOTSSL-1416

### CODE REVIEW

**Comment by gilles-peskine-arm** [comment]
@RonEld done

File: ChangeLog [hunk 1]
```diff
@@ -1,5 +1,10 @@
 mbed TLS ChangeLog (Sorted per branch, date)
 
+= mbed TLS 2.x.x branch released 2017-xx-xx
+
+Changes
+   * Clarified the documentation of mbedtls_ssl_setup.
```
Comment (hanno-arm) [inline ChangeLog]: Afaik, we usually write the ChangeLog in imperative style.

**Comment by hanno-arm** [comment]
@gilles-peskine-arm Did you also provide a backport to 2.7?

**Comment by gilles-peskine-arm** [comment]
@hanno-arm This PR is old enough that it can be merged into 2.7 just as well as development. The 2.1 backport is #937.

File: ChangeLog (hunk 1 above)
Comment (gilles-peskine-arm) [reply ChangeLog]: Since the last commit was a changelog, I force-pushed an amendment of the last commit with “Clarify” instead of ”Clarified“.

**Pull request merged**